plugins/module_utils/network/sonic/config/users/users.py

```diff
@@ -201,6 +199,26 @@ def _state_deleted(self, want, have, diff):
 
         return commands, requests
 
+    def _state_overridden(self, want, have, diff):
+        """ The command generator when state is overridden
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :param diff: the difference between want and have
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        # Delete all users except admin
+        commands = have
+        requests = self.get_delete_users_requests(commands, have)
+        send_requests(self._module, requests)
+
+        # Merge want configuration
+        commands = want
+        commands, requests = self._state_merged(want, have, commands)
+
+        return commands, requests
```

Comment:
Addressed.

Because of password not being in config I thought idempotency couldn't be handled but realized there's logic in diff to handle that. In the changes I pushed I set have to [] after deleting so I went ahead and sent the delete requests.